api/error_handler.py

```
@@ -0,0 +1,350 @@
+        
+        try:
+            # Get the current frame and caller information
+            current_frame = inspect.currentframe()
+            if current_frame and current_frame.f_back and current_frame.f_back.f_back:
+                caller_frame = current_frame.f_back.f_back
+                context["current_function"] = caller_frame.f_code.co_name
+                context["module"] = caller_frame.f_globals.get("__name__", "unknown")
+                
+                if caller_frame.f_back:
+                    context["calling_function"] = caller_frame.f_back.f_code.co_name
+        except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)